api/integration_alert_channels_datadog_test.go

```diff
@@ -0,0 +1,260 @@
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package api_test
+
+import (
+	"fmt"
+	"net/http"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/lacework/go-sdk/api"
+	"github.com/lacework/go-sdk/internal/intgguid"
+	"github.com/lacework/go-sdk/internal/lacework"
+)
+
+func TestIntegrationsNewDatadogAlertChannel(t *testing.T) {
+	subject := api.NewDatadogAlertChannel("integration_name",
+		api.DatadogChannelData{
+			DatadogSite: "eu",
+			DatadogType: "Events Summary",
+			ApiKey:      "datadog-key",
+		},
+	)
+	assert.Equal(t, api.DatadogChannelIntegration.String(), subject.Type)
+}
+
+func TestIntegrationsCreateDatadogAlertChannel(t *testing.T) {
+	var (
+		intgGUID   = intgguid.New()
+		fakeServer = lacework.MockServer()
+	)
+	fakeServer.MockAPI("external/integrations", func(w http.ResponseWriter, r *http.Request) {
+		assert.Equal(t, "POST", r.Method, "CreateDatadogAlertChannel should be a POST method")
+
+		if assert.NotNil(t, r.Body) {
+			body := httpBodySniffer(r)
+			assert.Contains(t, body, "integration_name", "integration name is missing")
+			assert.Contains(t, body, "DATADOG", "wrong integration type")
+			assert.Contains(t, body, "eu", "wrong datadog site")
+			assert.Contains(t, body, "Events Summary", "wrong datadog type")
+			assert.Contains(t, body, "datadog-key", "wrong datadog api key")
+			assert.Contains(t, body, "ENABLED\":1", "integration is not enabled")
+		}
+
+		fmt.Fprintf(w, datadogChannelIntegrationJsonResponse(intgGUID))
+	})
+	defer fakeServer.Close()
+
+	c, err := api.NewClient("test",
+		api.WithToken("TOKEN"),
+		api.WithURL(fakeServer.URL()),
+	)
+	assert.Nil(t, err)
+
+	data := api.NewDatadogAlertChannel("integration_name",
+		api.DatadogChannelData{
+			DatadogSite: "eu",
+			DatadogType: "Events Summary",
+			ApiKey:      "datadog-key",
+		},
+	)
+
+	assert.Equal(t, "integration_name", data.Name, "Datadog integration name mismatch")
+	assert.Equal(t, "DATADOG", data.Type, "a new Datadog integration should match its type")
+	assert.Equal(t, 1, data.Enabled, "a new Datadog integration should be enabled")
+
+	response, err := c.Integrations.CreateDatadogAlertChannel(data)
+
+	assert.Nil(t, err)
+	assert.NotNil(t, response)
+	assert.True(t, response.Ok)
+	if assert.Equal(t, 1, len(response.Data)) {
+		resData := response.Data[0]
+		assert.Equal(t, intgGUID, resData.IntgGuid)
+		assert.Equal(t, "integration_name", resData.Name)
+		assert.True(t, resData.State.Ok)
+		assert.Equal(t, "eu", resData.Data.DatadogSite)
+		assert.Equal(t, "Events Summary", resData.Data.DatadogType)
```

Comment:
If we make `DatadogType` an enum, we could add a test that checks that if the user passes any invalid value, we return a helpful message. (only if we change the field type)